src/main/java/org/jenkinsci/plugins/ghprb/extensions/build/GhprbCancelBuildsOnUpdate.java

```diff
@@ -0,0 +1,113 @@
+package org.jenkinsci.plugins.ghprb.extensions.build;
+
+import java.util.logging.Logger;
+
+import org.jenkinsci.plugins.ghprb.Ghprb;
+import org.jenkinsci.plugins.ghprb.GhprbCause;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbBuildStep;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbExtension;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbExtensionDescriptor;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbGlobalDefault;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbGlobalExtension;
+import org.jenkinsci.plugins.ghprb.extensions.GhprbProjectExtension;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.model.Queue;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.util.RunList;
+import jenkins.model.Jenkins;
+
+public class GhprbCancelBuildsOnUpdate extends GhprbExtension implements GhprbBuildStep, GhprbGlobalExtension, GhprbProjectExtension, GhprbGlobalDefault {
+
+    @Extension
+    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
+    private static final Logger logger = Logger.getLogger(GhprbCancelBuildsOnUpdate.class.getName());
+    
+    private final Boolean overrideGlobal;
+
+    @DataBoundConstructor
+    public GhprbCancelBuildsOnUpdate(Boolean overrideGlobal) {
+        this.overrideGlobal = overrideGlobal;
+    }
+    
+    public Boolean getOverrideGlobal() {
+        return overrideGlobal == null ? false : overrideGlobal;
+    }
+
+    private void cancelCurrentBuilds(AbstractProject<?, ?> project,
+                                     Integer prId) {
+        if (overrideGlobal) {
+            return;
+        }
+        
+        Queue queue = Jenkins.getInstance().getQueue();
```

Comment:
Awesome.Okay then that's probably okay.  I'm guessing there are still probably some races here that may be unavoidable.  For instance, if I push two commits to a PR one after the other and the second set of jobs queues first, the older commit may cancel the newer commit's jobs.  

These instances are probably few and far between enough to not worry about too much though.